checkmarx-ltd/VSTS-Plugin
# Project Branch Support

**Changes**
Added support of branching project

**Test Cases** 

1. Create a new project without enableSAST branching - 
   Expected -Pipeline should be green - Working 
2. Run Scan with the existing project  without enableSAST branching 
   Expected -Pipeline should be green - Working 
3. Create a new project with enableSAST branching and masterbracnhing project is blank 
	Expected -will get error (Scan cannot be completed. Master branch project name is mandatory to create branched project.) - Working
4. Create new project with enableSAST branching and masterbracnhing project is invalid 
	Expected - Error (Scan cannot be completed. Master branch project does not exist: abckjdkjdldkhebfffklf) - working
5. Create new project with enableSAST branching and masterbracnhing project is valid one
	Expected - Pipeline should be green - CHeck SAST is project branched or not - working
6. Run Scan with existing project and enableSAST branching and masterbracnhing project is valid one
	Expected -Pipeline should be green but it will not update branched project because its already add branched project for same - Working  
7. Enable checkbox Deny and Create new project without enableSAST branching
8. Enable checkbox Deny and Create new project with enableSAST branching and masterbracnhing project is valid (Same expected result for 7 & 8 )
   Expected - Error (Scan cannot be completed. Creation of the new project [PB-Case-7] is not authorized. Please use an existing project. You can enable the creation of new projects by disabling the Deny new Checkmarx projects creation checkbox in the Checkmarx plugin global settings) - Working
9. Enable checkbox Deny  and Run Scan with existing project  without enableSAST branching
	Expected - Pipeline should be green - Working
